modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/io/JsonESDocumentWriter.java

```diff
@@ -212,7 +217,14 @@ protected static void writeProperties(JsonGenerator jg, DocumentModel doc, Strin
         }
 
         for (Property p : properties) {
-            writer.writeProperty(jg, p);
+            try {
+                writer.writeProperty(jg, p);
+            } catch (ClassCastException e) {
+                var nuxeoException = new NuxeoException(e);
+                nuxeoException.addInfo(
+                        String.format("writing JSON property failed on document: %s for property: %s", doc, p));
```

Comment:
You can do directly `throw new NuxeoException(message, e)`, no need for `addInfo` which is only useful if you rethrow the same `NuxeoException` that you caught.